src/platform/monitoring/DowntimeNotification/reducer/index.js

```diff
@@ -0,0 +1,51 @@
+import { createServiceMap } from '../util/helpers';
+
+import {
+  RECEIVE_SCHEDULED_DOWNTIME,
+  RETRIEVE_SCHEDULED_DOWNTIME,
+  DISMISS_DOWNTIME_APPROACHING_MODAL
+} from '../actions';
+
+const DISMISSED_DOWNTIME_WARNINGS = 'DISMISSED_DOWNTIME_WARNINGS';
+function getDismissedDowntimeApproachingModals() {
+  const rawData = window.sessionStorage[DISMISSED_DOWNTIME_WARNINGS];
+  return rawData ? JSON.parse(rawData) : [];
+}
+
+const initialState = {
+  isReady: false,
+  isPending: false,
+  serviceMap: null,
+  dismissedDowntimeApproachingModals: getDismissedDowntimeApproachingModals()
+};
+
+export default function scheduledDowntime(state = initialState, action) {
+  switch (action.type) {
+
+    case RECEIVE_SCHEDULED_DOWNTIME:
+      return {
+        ...state,
+        isReady: true,
+        isPending: false,
+        serviceMap: createServiceMap(action.data)
+      };
+
+    case RETRIEVE_SCHEDULED_DOWNTIME:
+      return {
+        ...state,
+        isPending: true
+      };
+
+    case DISMISS_DOWNTIME_APPROACHING_MODAL: {
+      const dismissedDowntimeApproachingModals = state.dismissedDowntimeApproachingModals.concat(action.appTitle);
+      window.sessionStorage[DISMISSED_DOWNTIME_WARNINGS] = JSON.stringify(dismissedDowntimeApproachingModals);
```

Comment:
Oh gotcha, that makes sense. I'll move that over. Thanks.

Reply:
I've been working on this, and I actually think this is an instance where it is cleanest to have it in the reducer. The initial state needs some way of initializing out of the session and it makes sense to me to have all of the session stuff in one place, especially since it should always be in sync with the state. Does that make sense to you too?

Reply:
I can see why you would want to do that, but it goes against how Redux defines what reducers do (https://redux.js.org/basics/reducers): 

> Given the same arguments, it should calculate the next state and return it. No surprises. No side effects. No API calls. No mutations. Just a calculation.

This is the main expectation of reducers, so I'm really, really hesitant to go against it.

Reply:
Side effects can go in the action creators.

Reply:
@jbalboni Okay, if you feel strongly that's what I will do. 

Appreciate it @rianfowler but @jbalboni actually explained that in his initial comment on this thread.

Reply:
We could also generalize the behavior into a middleware if we want to update session storage in many actions and keep side effects out of actions.

Reply:
@rianfowler That's a great idea, I was thinking something similar. I mentioned a little while back that it might be nice to see our session storage behavior centralized or cleaned up so that we know exactly what we are storing there.